generator/generators/GoGenerator.py

```diff
@@ -0,0 +1,93 @@
+'''
+ Copyright (c) 2017 HERE Europe B.V.
+ See the LICENSE file in the root of this project for license details.
+'''
+from generator.tree.nodes.archive import Archive
+from generator.tree.nodes.node import Node
+from generator.tree.nodes.resources import Instance, Vector, Multivector, RawData
+from generator.tree.nodes.resources.archive import Archive as ArchiveResource
+from generator.tree.nodes.trivial import Structure, Constant
+
+from .BaseGenerator import BaseGenerator
+
+
+class GoGenerator(BaseGenerator):
+    def __init__(self):
+        BaseGenerator.__init__(self, "go/go.jinja2")
+
+    def _supported_nodes(self):
+        return [Structure, Archive, Constant]
+
+    def _populate_environment(self, env):
+        def _decorate_archive_type(tree, value):
+            assert isinstance(value, Node)
+            return value.name
+
+        def to_go_doc(value):
+            lines = value.doc.splitlines()
+            return '\n'.join(["// " + s for s in lines if len(s) != 0])
+
+        def type_mapping(type, struct):
+            if is_bool(type):
+                return "uint8"
+            else:
+                return go_mapping(type)
+
+        def type_mapping_with_bool(type):
```

Comment:
`type` is reserved keyword in Python. We should use something like `t`.

Reply:
Done